lib/nodes/image.js

```diff
@@ -0,0 +1,33 @@
+export const name = "image";
+
+export function schema(options) {
+  const images = options.images;
+  const imagesReverse = Object.fromEntries(
+    Object.entries(images).map(([k, v]) => [v, k]),
+  );
+  return {
+    group: "block",
+    attrs: { markdown: {} },
+    draggable: true,
+    parseDOM: [
+      {
+        tag: "img[src]",
+        getAttrs(dom) {
+          const src = dom.getAttribute("src");
+          const markdown = imagesReverse[src];
+          return markdown ? { markdown } : false;
+        },
+      },
+    ],
+    toDOM(node) {
+      return ["img", { src: images[node.attrs.markdown] }];
+    },
```

Comment:
This is more a comment on Prosemirror than the PR, but it feels a bit odd that very dynamic serialisation functions `parseDOM` and `toDOM` are part of the schema. If I'd been writing this library I'd have required those to be implemented as serialisers like the markdown serialiser. Ok yes, you're always going to want a DOM serialiser, but mixing the concerns of configuration and serialisation is quite painful. Having to pass the image paths injected by the user into schema config doesn't seem right to me